### PR TITLE
ui: claiming button improvements

### PR DESCRIPTION
**Pull request by karolina-siemieniuk-morawska**
* add visible but grayed out claiming button for not logged in users
* fix the unnecessary pluralisation of claim button items in literature item
* fix for claiming mixed papers on publication selection

* ref: cern-sis/issues-inspire#42

### CODE REVIEW

File: ui/src/literature/containers/LiteratureSearchContainer.jsx [hunk 1]
```diff
@@ -139,6 +144,7 @@ function LiteratureSearch({
                     <AssignAllDifferentProfileActionContainer />
                   )}
                 {assignNoProfileViewCondition && <AssignNoProfileAction />}
+                {!loggedIn && AUTHOR_PROFILE_VIEW.includes(namespace) && <ClaimingDisabledButton />}
```
Comment (drjova) [inline ui/src/literature/containers/LiteratureSearchContainer.jsx:147]: I would suggest to use a context as we do for the rest of the cases :)

File: ui/src/authors/containers/AssignAllOwnProfileActionContainer.jsx [hunk 2]
```diff
@@ -9,6 +9,11 @@ const stateToProps = (state) => ({
     state.authors.get('publicationSelectionClaimed').size > 0 &&
     state.authors.get('publicationSelectionUnclaimed').size === 0,
   numberOfSelected: state.authors.get('publicationSelection').size,
+  claimedPapersTooltip:
+    state.authors.get('publicationSelectionClaimed').size ===
+    state.authors.get('publicationSelection').size
+      ? 'All selected papers are already claimed!'
```
Comment (drjova) [inline ui/src/authors/containers/AssignAllOwnProfileActionContainer.jsx:15]: why we need this change? is not part of the requirements 🤔

File: ui/src/authors/containers/AssignOneActionContainer.jsx [hunk 3]
```diff
@@ -8,6 +8,11 @@ import {
 } from '../../actions/authors';
 import AssignAction from '../components/AssignAction';
 
+const stateToProps = () => ({
+  numberOfSelected: 1,
+  claimedPapersTooltip: 'This paper is already claimed',
+});
```
Comment (drjova) [inline ui/src/authors/containers/AssignOneActionContainer.jsx:14]: ditto, this is not part of the requirements 🤔

File: ui/src/authors/containers/AssignOneOwnProfileContainer.jsx [hunk 4]
```diff
@@ -11,6 +11,11 @@ import {
 } from '../../actions/authors';
 import AssignOwnProfileAction from '../components/AssignOwnProfileAction';
 
+const stateToProps = () => ({
+  numberOfSelected: 1,
+  claimedPapersTooltip: 'This paper is already claimed',
+});
+
```
Comment (drjova) [inline ui/src/authors/containers/AssignOneOwnProfileContainer.jsx:18]: ditto, this is not part of the requirements 🤔

**Comment by drjova** [comment]
@karolina-siemieniuk-morawska the second part of the requirements is missing:

> Similarly, when a user is logged in but doesn't have a profile, the button should be grayed out.

I'm a logged in user (without a profile) and I see the button:


```json
{
  "loggedIn": true,
  "isSigningUp": false,
  "signUpError": null,
  "isUpdatingOrcidPushSetting": false,
  "updateOrcidPushSettingError": null,
  "preferences": {
    "preferredCiteFormat@V2": "application/x-bibtex",
    "preferredCitationSummaryEnabling": false,
    "excludeSelfCitations": false
  },
  "data": {
    "email": "admin@inspirehep.net",
    "roles": [
      "superuser"
    ],
    "orcid": null,
    "allow_orcid_push": null
  }
}

```

Screenshot 2022-06-08 at 09 33 22

**Comment by karolina-siemieniuk-morawska** [comment]
@drjova that's because you are logged in as a superuser - for superusers and catalogers there is a special context provided that allows them to do all kind of actions (AssignViewContext.Provider). If you are logged in as a normal user with no profile 

```
data: {email: "johndoe@inspirehep.net", roles: [], orcid: null, allow_orcid_push: null}
allow_orcid_push: null
email: "johndoe@inspirehep.net"
orcid: null
roles: []
isSigningUp: false
isUpdatingOrcidPushSetting: false
loggedIn: true
preferences: {preferredCiteFormat@V2: "application/x-bibtex", preferredCitationSummaryEnabling: false,…}
excludeSelfCitations: false
preferredCitationSummaryEnabling: false
preferredCiteFormat@V2: "application/x-bibtex"
signUpError: null
updateOrcidPushSettingError: null
```

you will see the button like this:
Screenshot 2022-06-08 at 11 35 09

**Comment by drjova** [comment]
Thank you for the explanation.

**Review by drjova — APPROVED** [review]

**Pull request merged**